examples/src/main/scala/TracingExample.scala

```diff
@@ -44,11 +47,35 @@ object Work {
 }
 
 object TracingExample extends IOApp.Simple {
-  def tracerResource: Resource[IO, Tracer[IO]] =
-    Resource
-      .eval(IO(GlobalOpenTelemetry.get))
-      .evalMap(OtelJava.forSync[IO])
-      .evalMap(_.tracerProvider.tracer("Example").get)
+
+  // This would be a library function
+  def local[E](a: E): IO[Local[IO, E]] =
+    IOLocal(a).map { ioLocal =>
+      new Local[IO, E] {
+        override def local[A](fa: IO[A])(f: E => E): IO[A] =
+          ioLocal.get.flatMap(prev =>
+            ioLocal
+              .set(f(prev))
+              .bracket(Function.const(fa))(Function.const(ioLocal.set(prev)))
+          )
+        override val applicative: Applicative[IO] =
+          Applicative[IO]
+        override def ask[E2 >: E]: IO[E2] =
+          ioLocal.get
+      }
+    }
```

Comment:
This is my interpretation of https://github.com/typelevel/cats-effect/issues/3385, but I haven't tested it.